c2rust-analyze/src/c_void_casts.rs

```diff
@@ -274,16 +287,88 @@ impl<'tcx> CVoidCasts<'tcx> {
     /// [`Cast`]: rustc_middle::mir::Rvalue::Cast
     /// [`From`]: CVoidCastDirection::From
     /// [`To`]: CVoidCastDirection::To
-    pub fn should_skip_stmt(&self, stmt: &Statement<'tcx>) -> bool {
-        let (lhs, rv) = match get_assign_sides(stmt) {
-            None => return false,
-            Some(it) => it,
-        };
-        self.to.contains(lhs)
-            || self.from.contains(match get_cast_place(rv) {
-                None => return false,
-                Some(it) => it,
-            })
+    pub fn should_skip_stmt(&self, loc: &Location) -> bool {
+        self.to.contains(loc) || self.from.contains(loc)
+    }
+
+    fn is_place_local_modified(p: &Place, stmt: &Statement) -> bool {
```

Comment:
```suggestion
    fn is_place_unmodified_by_stmt(p: &Place, stmt: &Statement) -> bool {
```

I think this was meant to be "unmodified", right?  Since Stuart said it's
> unmodified only if (1) it's just a `Local`, with no projections, and
and here it's returning `false` when there is a projection.

Also, I think `is_place_unmodified_by_stmt` reads more clearly.

Reply:
93a30394, it's checking for modification, not unmodification